src/PluginClientBuilder.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Http\Client\Common;
+
+use Http\Client\HttpAsyncClient;
+use Psr\Http\Client\ClientInterface;
+
+/**
+ * Build an instance of a PluginClient with a dynamic list of plugins.
+ *
+ * @author Baptiste Clavié <clavie.b@gmail.com>
+ */
+final class PluginClientBuilder
+{
+    /** @var Plugin[][] */
+    private $plugins = [];
+
+    public function addPlugin(Plugin $plugin, int $priority = 0): self
```

Comment:
we should add a phpdoc about $priority, to say whether high or low numbers come first in the plugin chain.